InscryptionAPI/Card/CardExtensions.cs

```diff
@@ -296,61 +367,175 @@ public static CardInfo SetRare(this CardInfo info)
+                CardInfo creatureWithinCard = cards.CardByName(iceCubeName);
+
+                if (target != null && creatureWithinCard == null && target.IsOldApiCard()) // Maybe this is due to poor naming conventions allowed by the old API.
+                    creatureWithinCard = cards.CardByName($"{target.GetModPrefix()}_{iceCubeName}");
+
+                if (target != null && creatureWithinCard != null)
+                    target.SetIceCube(creatureWithinCard, mods);
+
+                return cards;
+            };
+        }
+        else
+        {
+            info.SetIceCube(creatureWithin, mods);
+        }
+
         return info;
     }
 
     /// <summary>
     /// Sets the tail parameters of the card. These parameters are used to make the TailOnHit ability function correctly.
     /// This function uses delayed loading to attach the tail to the card, so if the tail card doesn't exist yet, this function will still work.
     /// </summary>
+    /// <param name="info">Card to access</param>
     /// <param name="tailName">The name of the card that will be generated as the "tail" when the first hit is dodged.</param>
     /// <param name="pathToLostTailArt">The path to the .png file containing the lost tail artwork (relative to the Plugins directory)</param>
     /// <param name="mods">A set of card mods to be applied to the tail</param>
-    /// <returns></returns>
+    /// <returns>The same card info so a chain can continue</returns>
     public static CardInfo SetTail(this CardInfo info, string tailName, string pathToLostTailArt = null, IEnumerable<CardModificationInfo> mods = null)
     {
         Texture2D lostTailPortrait = pathToLostTailArt == null ? null : TextureHelper.GetImageAsTexture(pathToLostTailArt);
-        return info.SetTail(tailName, lostTailPortrait, mods:mods);
+        return info.SetTail(tailName, lostTailPortrait, mods: mods);
     }
 
     /// <summary>
     /// Sets the tail parameters of the card. These parameters are used to make the TailOnHit ability function correctly.
     /// </summary>
+    /// <param name="info">Card to access</param>
     /// <param name="tail">The card that will be generated as the "tail" when the first hit is dodged.</param>
     /// <param name="tailLostPortrait">The texture containing the card portrait</param>
     /// <param name="filterMode">The filter mode for the texture, or null if no change</param>
     /// <param name="mods">A set of card mods to be applied to the tail</param>
-    /// <returns></returns>
+    /// <returns>The same card info so a chain can continue</returns>
     public static CardInfo SetTail(this CardInfo info, CardInfo tail, Texture2D tailLostPortrait, FilterMode? filterMode = null, IEnumerable<CardModificationInfo> mods = null)
     {
-        info.tailParams = new();
-        info.tailParams.tail = tail;
+        info.tailParams = new()
+        {
+            tail = CardLoader.Clone(tail)
```

Comment:
This clone should only happen if the mods are not null; otherwise, just use the card